scu-de/src/fiskaltrust.Middleware.SCU.DE.Swissbit/SwissbitSCU.cs

```diff
@@ -784,6 +801,73 @@ private async Task CacheExportAsync(Guid exportId, string clientId = null, bool
             });
         }
 
+        private async Task CacheExportIncrementalAsync(Guid exportId, long startTransactionNr, long maxNumberOfRecords, long currentNumberOfTransactions)
+        {
+            var endTransactionNr = LastNumberOfRecords(startTransactionNr, maxNumberOfRecords, currentNumberOfTransactions);
+            long newStartTransactionNr = -1;
+
+            await _lockingHelper.PerformWithLock(_hwLock, async () =>
+            {
+                try
+                {
+                    using (var stream = new MemoryStream())
+                    {
+                        await GetProxy().ExportTarFilteredTransactionAsync(stream, (ulong) startTransactionNr, (ulong) endTransactionNr, null);
+                        stream.Position = 0;
+                        TarFileHelper.AppendTarStreamToTarFile(exportId.ToString(), stream);
+                        _logger.LogInformation($"Export total {currentNumberOfTransactions}. Partial Export from TransactionNr: {startTransactionNr} to TransactionNr: {endTransactionNr}");
+                    }
+                    newStartTransactionNr = GetLastExportedTransaction(exportId.ToString()) + 1;
+                }
+                catch (Exception ex)
+                {
+                    if (!ex.Message.Equals("Filtered Export: no matching entries, export would be empty. "))
+                    {
+                        _logger.LogError(ex, "Failed to execute {Operation} - TempFileName: {TempFileName}", nameof(CacheExportIncrementalAsync), exportId.ToString());
+                        SetExportState(exportId, ExportState.Failed, ex);
+                        throw;
+                    }
+                    _logger.LogInformation($"No Data: Export total {currentNumberOfTransactions}. Partial Export from TransactionNr: {startTransactionNr} to TransactionNr: {endTransactionNr}");
+
+                }
+            });
+
+            if (newStartTransactionNr == -1)
+            {
+                newStartTransactionNr = startTransactionNr + maxNumberOfRecords;
+            }
+            if (newStartTransactionNr <= currentNumberOfTransactions)
+            {
+                await CacheExportIncrementalAsync(exportId, newStartTransactionNr, maxNumberOfRecords, currentNumberOfTransactions).ConfigureAwait(false);
```

Comment:
If the `maxNumberOfRecords` is really small and the file really big we could possibly blow up the stack if we do it recursively.

We just need to remember that if it happens later on 😁

Reply:
thats why i would have been for hardcode 10.000 :)